{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[12/26/23 11:07:25] INFO     Kedro project paediatric_data       session.py:364\n",
      "Filepath: ...\n",
      "sftp://193.218.11.199/upload/Test_Data/pediatric_data/AIOLOS19-12-2023.xlsx\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "┌───────────────────── Traceback (most recent call last) ─────────────────────┐\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro\\io\\c │\n",
      "│ ore.py:161 in from_config                                                   │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\Downloads\\paediatric_data\\src\\paediatric_data_processing\\e │\n",
      "│ xtras\\datasets\\sftp_dataset.py:66 in __init__                               │\n",
      "│                                                                             │\n",
      "│    63 │   │   print(\"Filepath: ...\")                                        │\n",
      "│    64 │   │   print(filepath)                                               │\n",
      "│    65 │   │                                                                 │\n",
      "│ >  66 │   │   super().__init__(                                             │\n",
      "│    67 │   │   │   filepath, load_args, save_args, version, credentials, fs_ │\n",
      "│    68 │   │   )                                                             │\n",
      "│    69                                                                       │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro_data │\n",
      "│ sets\\pandas\\excel_dataset.py:169 in __init__                                │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\fsspec\\reg │\n",
      "│ istry.py:267 in filesystem                                                  │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\fsspec\\spe │\n",
      "│ c.py:79 in __call__                                                         │\n",
      "└─────────────────────────────────────────────────────────────────────────────┘\n",
      "TypeError: SFTPFileSystem.__init__() missing 1 required positional argument: \n",
      "'host'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "┌───────────────────── Traceback (most recent call last) ─────────────────────┐\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\runpy.py:196 in          │\n",
      "│ _run_module_as_main                                                         │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\runpy.py:86 in _run_code │\n",
      "│                                                                             │\n",
      "│ in <module>:7                                                               │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro\\fram │\n",
      "│ ework\\cli\\cli.py:211 in main                                                │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\click\\core │\n",
      "│ .py:1130 in __call__                                                        │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro\\fram │\n",
      "│ ework\\cli\\cli.py:139 in main                                                │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\click\\core │\n",
      "│ .py:1055 in main                                                            │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\click\\core │\n",
      "│ .py:1657 in invoke                                                          │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\click\\core │\n",
      "│ .py:1404 in invoke                                                          │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\click\\core │\n",
      "│ .py:760 in invoke                                                           │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro\\fram │\n",
      "│ ework\\cli\\project.py:453 in run                                             │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro\\fram │\n",
      "│ ework\\session\\session.py:417 in run                                         │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro\\fram │\n",
      "│ ework\\context\\context.py:277 in _get_catalog                                │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro\\io\\d │\n",
      "│ ata_catalog.py:310 in from_config                                           │\n",
      "│                                                                             │\n",
      "│ C:\\Users\\tfauvel\\.conda\\envs\\kedro-environment\\lib\\site-packages\\kedro\\io\\c │\n",
      "│ ore.py:163 in from_config                                                   │\n",
      "└─────────────────────────────────────────────────────────────────────────────┘\n",
      "DatasetError: \n",
      "SFTPFileSystem.__init__() missing 1 required positional argument: 'host'.\n",
      "Dataset 'PAEDIATRIC_SOURCE' must only contain arguments valid for the \n",
      "constructor of \n",
      "'paediatric_data_processing.extras.datasets.sftp_dataset.SFTPExcelDataSet'.\n"
     ]
    }
   ],
   "source": [
    "!kedro run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "from src.paediatric_data_processing.extras.datasets.mongodb_dataset import MongoDBDataSet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys \n",
    "sys.path.insert(0, \".\")\n",
    " \n",
    "from src.paediatric_data_processing.extras.datasets.mongodb_dataset import MongoDBDataSet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!kedro run --env=dev --docker-args=\"--env AWS_ACCESS_KEY_ID=\"AKIA3EYHIGTO4PRWJNJT\" --env AWS_SECRET_ACCESS_KEY=\"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\" --env MongoDBUsername=\"QuintenPipelineUser\" --env MongoDBPassword=\"PDb6uqwogle8sYFL\" --env CGM_PASSWORD=\"cmxJwnSmzPnQa55\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "docker run -v 'C:\\Users\\tfauvel\\Downloads\\paediatric_data\\conf\\local:/home/kedro_docker/conf/local' -v 'C:\\Users\\tfauvel\\Downloads\\paediatric_data\\data:/home/kedro_docker/data' -v 'C:\\Users\\tfauvel\\Downloads\\paediatric_data\\logs:/home/kedro_docker/logs' -v 'C:\\Users\\tfauvel\\Downloads\\paediatric_data\\notebooks:/home/kedro_docker/notebooks' -v 'C:\\Users\\tfauvel\\Downloads\\paediatric_data\\references:/home/kedro_docker/references' -v 'C:\\Users\\tfauvel\\Downloads\\paediatric_data\\results:/home/kedro_docker/results' --rm --name paediatric_data-run --env AWS_ACCESS_KEY_ID=AKIA3EYHIGTO4PRWJNJT --env AWS_SECRET_ACCESS_KEY=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 --env MongoDBUsername=QuintenPipelineUser --env MongoDBPassword=PDb6uqwogle8sYFL --env CGM_PASSWORD=cmxJwnSmzPnQa55 paediatric_data kedro run --env=dev\n",
      "[12/26/23 09:22:44] WARNING  /usr/local/lib/python3.10/site-pack warnings.py:109\n",
      "                             ages/kedro_datasets/json/json_datas                \n",
      "                             et.py:10: KedroDeprecationWarning:                 \n",
      "                             'AbstractVersionedDataSet' has been                \n",
      "                             renamed to                                         \n",
      "                             'AbstractVersionedDataset', and the                \n",
      "                             alias will be removed in Kedro                     \n",
      "                             0.19.0                                             \n",
      "                               from kedro.io.core import (                      \n",
      "                                                                                \n",
      "                    WARNING  /usr/local/lib/python3.10/site-pack warnings.py:109\n",
      "                             ages/kedro_datasets/json/json_datas                \n",
      "                             et.py:10: KedroDeprecationWarning:                 \n",
      "                             'DataSetError' has been renamed to                 \n",
      "                             'DatasetError', and the alias will                 \n",
      "                             be removed in Kedro 0.19.0                         \n",
      "                               from kedro.io.core import (                      \n",
      "                                                                                \n",
      "                    WARNING  /usr/local/lib/python3.10/site-pack warnings.py:109\n",
      "                             ages/kedro_datasets/matplotlib/matp                \n",
      "                             lotlib_writer.py:12:                               \n",
      "                             KedroDeprecationWarning:                           \n",
      "                             'AbstractVersionedDataSet' has been                \n",
      "                             renamed to                                         \n",
      "                             'AbstractVersionedDataset', and the                \n",
      "                             alias will be removed in Kedro                     \n",
      "                             0.19.0                                             \n",
      "                               from kedro.io.core import (                      \n",
      "                                                                                \n",
      "                    WARNING  /usr/local/lib/python3.10/site-pack warnings.py:109\n",
      "                             ages/kedro_datasets/matplotlib/matp                \n",
      "                             lotlib_writer.py:12:                               \n",
      "                             KedroDeprecationWarning:                           \n",
      "                             'DataSetError' has been renamed to                 \n",
      "                             'DatasetError', and the alias will                 \n",
      "                             be removed in Kedro 0.19.0                         \n",
      "                               from kedro.io.core import (                      \n",
      "                                                                                \n",
      "[12/26/23 09:22:45] WARNING  /usr/local/lib/python3.10/site-pack warnings.py:109\n",
      "                             ages/kedro_datasets/plotly/json_dat                \n",
      "                             aset.py:10:                                        \n",
      "                             KedroDeprecationWarning:                           \n",
      "                             'AbstractVersionedDataSet' has been                \n",
      "                             renamed to                                         \n",
      "                             'AbstractVersionedDataset', and the                \n",
      "                             alias will be removed in Kedro                     \n",
      "                             0.19.0                                             \n",
      "                               from kedro.io.core import (                      \n",
      "                                                                                \n",
      "                    WARNING  /usr/local/lib/python3.10/site-pack warnings.py:109\n",
      "                             ages/kedro_datasets/tracking/metric                \n",
      "                             s_dataset.py:9:                                    \n",
      "                             KedroDeprecationWarning:                           \n",
      "                             'DataSetError' has been renamed to                 \n",
      "                             'DatasetError', and the alias will                 \n",
      "                             be removed in Kedro 0.19.0                         \n",
      "                               from kedro.io.core import                        \n",
      "                             DataSetError, get_filepath_str                     \n",
      "                                                                                \n",
      "                    WARNING  /usr/local/lib/python3.10/site-pack warnings.py:109\n",
      "                             ages/kedro_datasets/tracking/json_d                \n",
      "                             ataset.py:7:                                       \n",
      "                             KedroDeprecationWarning:                           \n",
      "                             'DataSetError' has been renamed to                 \n",
      "                             'DatasetError', and the alias will                 \n",
      "                             be removed in Kedro 0.19.0                         \n",
      "                               from kedro.io.core import                        \n",
      "                             DataSetError                                       \n",
      "                                                                                \n",
      "[12/26/23 09:22:45] INFO     Kedro project kedro_docker           session.py:365\n",
      "Filepath: ...\n",
      "sftp://193.218.11.199/upload/Test_Data/hcl_data/AIOLOS19-12-2023.xlsx\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "╭───────────────────── Traceback (most recent call last) ──────────────────────╮\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro/io/core.py:162 in from_config  │\n",
      "│                                                                              │\n",
      "│   159 │   │   │   ) from exc                                                 │\n",
      "│   160 │   │                                                                  │\n",
      "│   161 │   │   try:                                                           │\n",
      "│ ❱ 162 │   │   │   data_set = class_obj(**config)  # type: ignore             │\n",
      "│   163 │   │   except TypeError as err:                                       │\n",
      "│   164 │   │   │   raise DatasetError(                                        │\n",
      "│   165 │   │   │   │   f\"\\n{err}.\\nDataset '{name}' must only contain argumen │\n",
      "│                                                                              │\n",
      "│ /home/kedro_docker/src/paediatric_data_processing/extras/datasets/sftp_datas │\n",
      "│ et.py:26 in __init__                                                         │\n",
      "│                                                                              │\n",
      "│   23 │   │   print(\"Filepath: ...\")                                          │\n",
      "│   24 │   │   print(filepath)                                                 │\n",
      "│   25 │   │                                                                   │\n",
      "│ ❱ 26 │   │   super().__init__(                                               │\n",
      "│   27 │   │   │   filepath, load_args, save_args, version, credentials, fs_ar │\n",
      "│   28 │   │   )                                                               │\n",
      "│   29                                                                         │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro_datasets/pandas/excel_dataset. │\n",
      "│ py:169 in __init__                                                           │\n",
      "│                                                                              │\n",
      "│   166 │   │                                                                  │\n",
      "│   167 │   │   self._protocol = protocol                                      │\n",
      "│   168 │   │   self._storage_options = {**_credentials, **_fs_args}           │\n",
      "│ ❱ 169 │   │   self._fs = fsspec.filesystem(self._protocol, **self._storage_o │\n",
      "│   170 │   │                                                                  │\n",
      "│   171 │   │   self.metadata = metadata                                       │\n",
      "│   172                                                                        │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/fsspec/registry.py:291 in filesystem │\n",
      "│                                                                              │\n",
      "│   288 │   │   )                                                              │\n",
      "│   289 │                                                                      │\n",
      "│   290 │   cls = get_filesystem_class(protocol)                               │\n",
      "│ ❱ 291 │   return cls(**storage_options)                                      │\n",
      "│   292                                                                        │\n",
      "│   293                                                                        │\n",
      "│   294 def available_protocols():                                             │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/fsspec/spec.py:80 in __call__        │\n",
      "│                                                                              │\n",
      "│     77 │   │   │   cls._latest = token                                       │\n",
      "│     78 │   │   │   return cls._cache[token]                                  │\n",
      "│     79 │   │   else:                                                         │\n",
      "│ ❱   80 │   │   │   obj = super().__call__(*args, **kwargs)                   │\n",
      "│     81 │   │   │   # Setting _fs_token here causes some static linters to co │\n",
      "│     82 │   │   │   obj._fs_token_ = token                                    │\n",
      "│     83 │   │   │   obj.storage_args = args                                   │\n",
      "╰──────────────────────────────────────────────────────────────────────────────╯\n",
      "TypeError: SFTPFileSystem.__init__() missing 1 required positional argument: \n",
      "'host'\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "╭───────────────────── Traceback (most recent call last) ──────────────────────╮\n",
      "│ /usr/local/bin/kedro:8 in <module>                                           │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro/framework/cli/cli.py:211 in    │\n",
      "│ main                                                                         │\n",
      "│                                                                              │\n",
      "│   208 │   \"\"\"                                                                │\n",
      "│   209 │   _init_plugins()                                                    │\n",
      "│   210 │   cli_collection = KedroCLI(project_path=Path.cwd())                 │\n",
      "│ ❱ 211 │   cli_collection()                                                   │\n",
      "│   212                                                                        │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/click/core.py:1157 in __call__       │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro/framework/cli/cli.py:139 in    │\n",
      "│ main                                                                         │\n",
      "│                                                                              │\n",
      "│   136 │   │   )                                                              │\n",
      "│   137 │   │                                                                  │\n",
      "│   138 │   │   try:                                                           │\n",
      "│ ❱ 139 │   │   │   super().main(                                              │\n",
      "│   140 │   │   │   │   args=args,                                             │\n",
      "│   141 │   │   │   │   prog_name=prog_name,                                   │\n",
      "│   142 │   │   │   │   complete_var=complete_var,                             │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/click/core.py:1078 in main           │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/click/core.py:1688 in invoke         │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/click/core.py:1434 in invoke         │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/click/core.py:783 in invoke          │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro/framework/cli/project.py:453   │\n",
      "│ in run                                                                       │\n",
      "│                                                                              │\n",
      "│   450 │   with KedroSession.create(                                          │\n",
      "│   451 │   │   env=env, conf_source=conf_source, extra_params=params          │\n",
      "│   452 │   ) as session:                                                      │\n",
      "│ ❱ 453 │   │   session.run(                                                   │\n",
      "│   454 │   │   │   tags=tag,                                                  │\n",
      "│   455 │   │   │   runner=runner(is_async=is_async),                          │\n",
      "│   456 │   │   │   node_names=node_names,                                     │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro/framework/session/session.py:4 │\n",
      "│ 18 in run                                                                    │\n",
      "│                                                                              │\n",
      "│   415 │   │   │   \"runner\": getattr(runner, \"__name__\", str(runner)),        │\n",
      "│   416 │   │   }                                                              │\n",
      "│   417 │   │                                                                  │\n",
      "│ ❱ 418 │   │   catalog = context._get_catalog(  # noqa: protected-access      │\n",
      "│   419 │   │   │   save_version=save_version,                                 │\n",
      "│   420 │   │   │   load_versions=load_versions,                               │\n",
      "│   421 │   │   )                                                              │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro/framework/context/context.py:2 │\n",
      "│ 77 in _get_catalog                                                           │\n",
      "│                                                                              │\n",
      "│   274 │   │   )                                                              │\n",
      "│   275 │   │   conf_creds = self._get_config_credentials()                    │\n",
      "│   276 │   │                                                                  │\n",
      "│ ❱ 277 │   │   catalog = settings.DATA_CATALOG_CLASS.from_config(             │\n",
      "│   278 │   │   │   catalog=conf_catalog,                                      │\n",
      "│   279 │   │   │   credentials=conf_creds,                                    │\n",
      "│   280 │   │   │   load_versions=load_versions,                               │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro/io/data_catalog.py:311 in      │\n",
      "│ from_config                                                                  │\n",
      "│                                                                              │\n",
      "│   308 │   │   │   │   ds_layer = ds_config.pop(\"layer\", None)                │\n",
      "│   309 │   │   │   │   if ds_layer is not None:                               │\n",
      "│   310 │   │   │   │   │   layers[ds_layer].add(ds_name)                      │\n",
      "│ ❱ 311 │   │   │   │   data_sets[ds_name] = AbstractDataset.from_config(      │\n",
      "│   312 │   │   │   │   │   ds_name, ds_config, load_versions.get(ds_name), sa │\n",
      "│   313 │   │   │   │   )                                                      │\n",
      "│   314 │   │   dataset_layers = layers or None                                │\n",
      "│                                                                              │\n",
      "│ /usr/local/lib/python3.10/site-packages/kedro/io/core.py:164 in from_config  │\n",
      "│                                                                              │\n",
      "│   161 │   │   try:                                                           │\n",
      "│   162 │   │   │   data_set = class_obj(**config)  # type: ignore             │\n",
      "│   163 │   │   except TypeError as err:                                       │\n",
      "│ ❱ 164 │   │   │   raise DatasetError(                                        │\n",
      "│   165 │   │   │   │   f\"\\n{err}.\\nDataset '{name}' must only contain argumen │\n",
      "│   166 │   │   │   │   f\"constructor of '{class_obj.__module__}.{class_obj.__ │\n",
      "│   167 │   │   │   ) from err                                                 │\n",
      "╰──────────────────────────────────────────────────────────────────────────────╯\n",
      "DatasetError: \n",
      "SFTPFileSystem.__init__() missing 1 required positional argument: 'host'.\n",
      "Dataset 'PAEDIATRIC_SOURCE' must only contain arguments valid for the \n",
      "constructor of \n",
      "'paediatric_data_processing.extras.datasets.sftp_dataset.SFTPDataSet'.\n"
     ]
    }
   ],
   "source": [
    "!kedro docker run --env=dev --docker-args=\"--env AWS_ACCESS_KEY_ID=\"AKIA3EYHIGTO4PRWJNJT\" --env AWS_SECRET_ACCESS_KEY=\"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\" --env MongoDBUsername=\"QuintenPipelineUser\" --env MongoDBPassword=\"PDb6uqwogle8sYFL\" --env CGM_PASSWORD=\"cmxJwnSmzPnQa55\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.19.1'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import kedro\n",
    "kedro.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2.0.0'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import kedro_datasets\n",
    "kedro_datasets.__version__"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "kedro-environment",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
